### PR TITLE
Features/add health check

### PR DESCRIPTION
**Pull request by gareth72evans**
## UKHO.SalesCatalogueStub.Api.IntegrationTests

- HealthcheckTest.cs: Added New Project.  New Unit Test for Health Check Integration Test

### CODE REVIEW

**Review by bonpoole — APPROVED** [review]

File: src/UKHO.SalesCatalogueStub.Api.IntegrationTests/UKHO.SalesCatalogueStub.Api.IntegrationTests.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+
+    <IsPackable>false</IsPackable>
```
Comment (benbhall) [inline src/UKHO.SalesCatalogueStub.Api.IntegrationTests/UKHO.SalesCatalogueStub.Api.IntegrationTests.csproj:6]:
```suggestion

```

File: src/UKHO.SalesCatalogueStub.Api.IntegrationTests/HealthcheckTest.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        public async Task Test_HealthcheckTest()
+        {
+            var healthcheckUrl = new Uri("https://localhost:44325/health", UriKind.Absolute);
+            var isSuccess = false;
+            var returnedValue = "";
+
+            using (var httpClient = new HttpClient())
+            {
+
+                var expectedResult = await httpClient.GetAsync(healthcheckUrl.ToString());
```
Comment (benbhall) [inline src/UKHO.SalesCatalogueStub.Api.IntegrationTests/HealthcheckTest.cs:20]:
```suggestion
                var expectedResult = await httpClient.GetAsync(healthcheckUrl);
```
No need for a string convert as there is a `GetAsync(Uri)` overload.

File: src/UKHO.SalesCatalogueStub.Api.IntegrationTests/HealthcheckTest.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    public class HealthcheckTest
+    {
+        [Test]
+        public async Task Test_HealthcheckTest()
+        {
+            var healthcheckUrl = new Uri("https://localhost:44325/health", UriKind.Absolute);
+            var isSuccess = false;
+            var returnedValue = "";
+
+            using (var httpClient = new HttpClient())
```
Comment (benbhall) [inline src/UKHO.SalesCatalogueStub.Api.IntegrationTests/HealthcheckTest.cs:17]: Before this test suite grows it is worth switching over to `IHttpClientFactory`. See link for the history behind HttpClient use.

https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

Can leave it today though.

**Review by benbhall — APPROVED** [review]

**Review by bonpoole — APPROVED** [review]

File: src/UKHO.SalesCatalogueStub.Api.IntegrationTests/HealthcheckTest.cs (hunk 3 above)
Comment (SamirHassoun) [reply src/UKHO.SalesCatalogueStub.Api.IntegrationTests/HealthcheckTest.cs:17]: its just a simle healthcheck test, so one test is needed and not worth going for that at the moment

**Pull request merged**